src/containers/SemesterContainer.jsx

```diff
@@ -0,0 +1,29 @@
+import { connect } from 'react-redux';
+import Semester from '../components/Semester';
+
+const filterCourse = (semester, completedCourses, plannedCourses) => {
+  const filteredCompleted = completedCourses.filter(course =>
+    course.semester === semester,
+  );
+  const filteredPlanned = plannedCourses.filter(course =>
+    course.semester === semester,
+  );
+  return filteredPlanned.concat(filteredCompleted);
```

Comment:
Yeah, there's one case of that now in the database (see semester 1 of screenshot), although I think that was a mistake when it came to assigning the correct semester to it in the DB. I suppose if a student _planned_ to take a course, but took something else instead then they might have some extra junk floating around though...